### PR TITLE
Bugfix/plugin 349

### PR DESCRIPTION
**Pull request by bajram-adapt**

### CODE REVIEW

File: src/main/java/io/cdap/plugin/gcp/bigquery/sink/AbstractBigQuerySink.java [hunk 1]
```diff
@@ -112,7 +112,7 @@ public final void prepareRun(BatchSinkContext context) throws Exception {
     baseConfiguration.set(BigQueryConstants.CONFIG_JOB_ID, jobId);
     if (!context.isPreviewEnabled()) {
       createResources(bigQuery, GCPUtils.getStorage(project, credentials), config.getDataset(), bucket,
-                      config.getLocation(), cmekKey);
+        config.getLocation(), cmekKey);
```
Comment (CuriousVini) [inline src/main/java/io/cdap/plugin/gcp/bigquery/sink/AbstractBigQuerySink.java]: this change is not needed

File: src/main/java/io/cdap/plugin/gcp/bigquery/sink/AbstractBigQuerySink.java [hunk 2]
```diff
@@ -192,20 +192,20 @@ protected final GenericRecord toAvroRecord(StructuredRecord record) throws IOExc
   /**
    * Initializes output along with lineage recording for given table and its schema.
    *
-   * @param context batch sink context
-   * @param bigQuery big query client for the configured project
-   * @param outputName output name
-   * @param tableName table name
+   * @param context     batch sink context
```
Comment (CuriousVini) [inline src/main/java/io/cdap/plugin/gcp/bigquery/sink/AbstractBigQuerySink.java]: unintentional? please undo these changes

File: src/main/java/io/cdap/plugin/gcp/bigquery/sink/AbstractBigQuerySink.java [hunk 3]
```diff
@@ -311,6 +311,33 @@ private String configureBucket() {
...
+      // BQ will overwrite the schema for normal tables when write disposition is WRITE_TRUNCATE
+      //note - If write to single partition is supported in future, schema validation will be necessary
+      return;
+    }
+    FieldList bqFields = bqSchema.getFields();
+    List<Schema.Field> outputSchemaFields = Objects.requireNonNull(tableSchema.getFields());
+
+    List<String> remainingBQFields = BigQueryUtil.getBqFieldsMinusSchema(bqFields, outputSchemaFields);
+    for (String field : remainingBQFields) {
+      if (bqFields.get(field).getMode() != Field.Mode.NULLABLE) {
```
Comment (CuriousVini) [inline src/main/java/io/cdap/plugin/gcp/bigquery/sink/AbstractBigQuerySink.java:333]: we should also check if the types are same

**Review by CuriousVini — CHANGES_REQUESTED** [review]

File: src/main/java/io/cdap/plugin/gcp/bigquery/sink/AbstractBigQuerySink.java (hunk 2 above)
Comment (bajram-adapt) [reply src/main/java/io/cdap/plugin/gcp/bigquery/sink/AbstractBigQuerySink.java]: yes unintentional, I'll revert formatting back

File: src/main/java/io/cdap/plugin/gcp/bigquery/sink/AbstractBigQuerySink.java (hunk 3 above)
Comment (bajram-adapt) [reply src/main/java/io/cdap/plugin/gcp/bigquery/sink/AbstractBigQuerySink.java:333]: added check for Schema

File: pom.xml [hunk 4]
```diff
@@ -72,7 +72,7 @@
     <bigquery.connector.hadoop2.version>hadoop2-1.0.0</bigquery.connector.hadoop2.version>
     <commons.codec.version>1.4</commons.codec.version>
     <cdap.version>6.3.0-SNAPSHOT</cdap.version>
-    <cdap.plugin.version>2.4.0-SNAPSHOT</cdap.plugin.version>
+    <cdap.plugin.version>2.5.0-SNAPSHOT</cdap.plugin.version>
```
Comment (bajram-adapt) [inline pom.xml:75]: What should the pom version for this be? live develop has 2.5.0-SNAPSHOT or should we ignore pom versions if they are different?
